From: Jeffrey Feng <galago1992@gmail.com>
Subject: 
  Enable Cairo subpixel rendering in poppler-glib.
  Thanks to Yichao Zhou for his subpixel patch of previous poppler-0.43.
---
git diff --stat
 glib/poppler-page.cc        | 18 ++++++++++++++++++
 glib/poppler-page.h         |  1 +
 2 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/glib/poppler-page.cc b/glib/poppler-page.cc
index 8b5b4da0..d28084fb 100644
--- a/glib/poppler-page.cc
+++ b/glib/poppler-page.cc
@@ -346,6 +346,14 @@ _poppler_page_render (PopplerPage      *page,
     page->text = new TextPage (false);
     output_dev->setTextPage (page->text);
   }
+  bool subpixel_suited = poppler_page_suit_subpixel_rendering (page);
+  if (!printing && subpixel_suited) {
+    output_dev->setAntialias(CAIRO_ANTIALIAS_SUBPIXEL);
+    cairo_set_source_rgb (cairo, 1., 1., 1.);
+    cairo_paint (cairo);
+  } else {
+    output_dev->setAntialias(CAIRO_ANTIALIAS_DEFAULT);
+  }
   /* NOTE: instead of passing -1 we should/could use cairo_clip_extents()
    * to get a bounding box */
   cairo_save (cairo);
@@ -2447,3 +2455,13 @@ poppler_page_get_text_attributes_for_area (PopplerPage      *page,
 
   return g_list_reverse(attributes);
 }
+
+gboolean
+poppler_page_suit_subpixel_rendering (PopplerPage        *page)
+{
+  CairoOutputDev *output_dev;
+  g_return_val_if_fail (POPPLER_IS_PAGE (page), FALSE);
+
+  output_dev = page->document->output_dev;
+  return page->page->suitSubpixelRendering(output_dev);
+}
diff --git a/glib/poppler-page.h b/glib/poppler-page.h
index f99f0920..f4b5692b 100644
--- a/glib/poppler-page.h
+++ b/glib/poppler-page.h
@@ -147,6 +147,7 @@ void                   poppler_page_free_text_attributes (GList              *li
 POPPLER_PUBLIC
 GList *        poppler_page_get_text_attributes_for_area (PopplerPage        *page,
                                                           PopplerRectangle   *area);
+gboolean         poppler_page_suit_subpixel_rendering (PopplerPage        *page);
 
 /* A rectangle on a page, with coordinates in PDF points. */
 #define POPPLER_TYPE_RECTANGLE             (poppler_rectangle_get_type ())
